### DISCUSSION TITLE
USE_ADC_VCC with ESP32 (LolinD32)

### DISCUSSION THREAD

**Discussion by dannystettler** [arendst/Tasmota#14274] [Q&A] [answered]
I'm a newbie and need some help. 

I've tried to flash an ESP32 (lolind32) with Tasmota and because it should be battery powered, I would like to monitor the battery voltage. For the creation of the custom firmware I'm using Gitpod and modified the my_user_config.h file to enable the USE_ADC_VCC feature. Then I've compiled the firmware with the command "pio run -e tasmota32" and flashed the  output onto the ESP32. 

The flashing was successful and I could base-configure the ESP. However,  after I booted the ESP32 with batteries, I could not see any indication in the logs saying anything about the voltage level.....any hints?

**Comment by barbudor** [comment, marked as answer]
1) Tasmota support this feature only on ESP8266

2) This feature is generally useless because it measure the voltage at the VCC pin of the ESP. When using a LiIon/LiPo battery with a max voltage of 4.2V, there is always a 3V3 voltage regulator after the battery to feed the VCC. So you measure of VCC will see 3V3 continusouly until the battery is below ~3.5...3.3V (depending on the voltage regulator used on your board). So you will get a warning only when it's too late as your battery is close to the end (LiIon drops very quickly once below ~3.4V)

The only use case where it could be used, is when using a LiFePO4 battery which max voltage is 3.6V connected directly to the ESP VCC pin without any voltage regulator. In that case it could make sense.

When using a LiIon/LiPo battery (and obviously with Tasmota32) you should use a standard ADC pin and a voltage divider.

Forget this useless feature of the ADC.

**Comment by dannystettler** [comment]
Thank you very much for the explanation. Indeed, my plan was to use a LiFePO4 Battery. So your suggestion is to use a standard pin for measuring the voltage of the battery?

**Reply by sfromis** [reply]
... with a voltage divider to get within range
